File: packages/snow_draw_core/lib/draw/reducers/interaction/interaction_transition.dart
import 'package:meta/meta.dart';

import '../../models/draw_state.dart';

/// Interaction state transition result.
@immutable
class InteractionTransition {
  const InteractionTransition({required this.nextState});

  /// State unchanged.
  const InteractionTransition.unchanged(DrawState state)
    : this(nextState: state);

  final DrawState nextState;
}

File: packages/snow_draw_rust_engine/src/draw/reducers/interaction/interaction_transition.rs
#![allow(dead_code)]

use crate::draw::models::draw_state::DrawState;

/// Interaction state transition result.
#[derive(Clone, Debug, PartialEq)]
pub struct InteractionTransition {
    /// The draw state after applying the interaction reducer step.
    pub next_state: DrawState,
}

impl InteractionTransition {
    /// Creates a transition with the provided next state.
    pub const fn new(next_state: DrawState) -> Self {
        Self { next_state }
    }

    /// Creates a transition that keeps the state unchanged.
    pub const fn unchanged(state: DrawState) -> Self {
        Self::new(state)
    }
}
